kinto_redis/permission.py

```diff
@@ -86,18 +88,27 @@ def get_object_permission_principals(self, object_id, permission):
         return self._decode_set(members)
 
     @wrap_redis_error
-    def get_accessible_objects(self, principals, bound_permissions=None):
+    def get_accessible_objects(self, principals, bound_permissions=None, with_children=True):
         principals = set(principals)
 
         if bound_permissions:
             keys = ['permission:%s:%s' % op for op in bound_permissions]
         else:
             keys = ['permission:*']
-
         perms_by_id = dict()
         for key_pattern in keys:
             matched = self._client.scan_iter(match=key_pattern)
             for key in matched:
+                if bound_permissions and not with_children:
+                    one_of = False
+                    for object_id, permission in bound_permissions:
+                        pattern = 'permission:%s:%s' % (object_id, permission)
+                        regexp = re.compile('^%s$' % pattern.replace('*', '[^/]+'))
```

Comment:
Well the key is coming from matched and then you want to see if it matches one or the other.
What would be good is to do the regexp on redis side but apparently it is not possible.